drizzlepac/runastrodriz.py

```diff
@@ -1726,6 +1733,44 @@ def update_active_wcs(filename, wcsname, logfile=None):
     if logfile:
         _updateTrlFile(logfile, update_msg)
 
+
+def confirm_aposteriori_hdrlets(filename, logfile=None):
+    """Confirm that all the a posteriori headerlets are valid, and remove any that are invalid."""
+    update_msg = ""
+    num_sci_ext, extname = util.count_sci_extensions(filename)
+    extname_list = [(extname, x + 1) for x in range(num_sci_ext)]
+
+    hdu = fits.open(filename, mode='update')
+    # Get a list of all HDRLET extensions
+    hdrlet_extensions = wcsutil.headerlet.get_extname_extver_list(hdu, sciext='HDRLET')
+    hdrlet_wcsnames = wcsutil.headerlet.get_headerlet_kw_names(hdu, 'wcsname')
+
```

Comment:
Just making sure...is the order of results returned by get_extname_extver_list and get_headerlet_kw_names guaranteed to be the same? It seems like it would be, but I thought it was worth asking.

Reply:
The first function call, ``get_extname_extver_list``, only returns the tuples of all HDRLET extensions that were found in the FITS file by looping over all extensions in the FITS file in order using ``for extn in hdu``.  The second function also loops over the extensions in the FITS file in the same way, only appending results to the output list for HDRLET extensions.  As a result, the results should remain in lock-step with each other.    Good question.